File: UI/Property Grid/Editors/MarkDownEditor.vb
Public Class MarkDownEditor : Inherits PgCodeEditor(Of GitProject)

    'Public Overrides ReadOnly Property Highlighter As CodeHighlighter = CodeHighlighter.CPP
    Public Overrides ReadOnly Property Highlighter As CodeHighlighter = New XshdHighlighter("MarkDown-Mode")
    Public Overrides ReadOnly Property ShowControlChars As Boolean? = False

    Protected Overrides Sub ConfigureEditor(Editor As CodeEditor, Instance As GitProject, pd As PropertyDescriptor)
        Editor.HighlightMode = CodeEditor.HighlightingMode.MarkDown
        'Editor.CompletionProvider = New CssCompletionProvider(Instance)
    End Sub

    Public Overrides Function GetCompletionProvider(Instance As Object, pd As PropertyDescriptor) As CompletionProvider
        ' Return New CssCompletionProvider(Instance)
    End Function
End Class

File: UI/Property Grid/Editors/MarkDownEditor.cs
using Grooper;
using System.ComponentModel;

namespace GrooperGit
{
    internal class MarkDownEditor : PgCodeEditor<GitProject>
    {
        public override CodeHighlighter Highlighter { get; } = new XshdHighlighter("MarkDown-Mode");
        public override bool? ShowControlChars { get; } = false;
        protected override void ConfigureEditor(CodeEditor Editor, GitProject Instance, PropertyDescriptor pd)
        {
            Editor.HighlightMode = CodeEditor.HighlightingMode.MarkDown;
        }
        public override CompletionProvider GetCompletionProvider(object instance, PropertyDescriptor pd)
        {
            // Uncomment and adjust as needed
            // return new CssCompletionProvider(instance);
            return null;
        }
    }
}
